src/Control/Bind.purs

```diff
@@ -63,6 +63,17 @@ infixr 1 bindFlipped as =<<
 instance bindFn :: Bind ((->) r) where
   bind m f x = f (m x) x
 
+-- | The `Array` monad's "do notation" works like a nested for loop. Each
```

Comment:
I would be more comfortable with an explanation which doesn't refer to do notation here - or, at least, start with one which doesn't refer to do notation, and then add the part about do notation afterwards. That way, at least a part of these docs will make sense to people who don't yet understand the do notation desugaring. I think it's common to use Array's Bind instance explicitly via `>>=` rather than within a `do` block as well, which is another reason to understand how Bind Array works outside of the context of do notation.

I think something similar to https://pursuit.purescript.org/packages/purescript-arrays/5.3.1/docs/Data.Array#v:concatMap would be good - although for prelude, if we can avoid referring to downstream libraries in the examples, that would be preferable. We could also add a note to `concatMap` saying it's the same as `bind`.

Reply:
Good point. I've updated that.